.github/workflows/python-app.yml

```
@@ -65,14 +65,19 @@ jobs:
         if [ -f test-requirements.txt ]; then pip install -r test-requirements.txt; fi
     - name: Type check with pyright
       run: pyright
+    - name: Run benchmarks
+      uses: CodSpeedHQ/action@v3
+      with:
+        token: ${{ secrets.CODSPEED_TOKEN }}
+        run: pytest tests/ --codspeed
```

Hey @pcriadoperez, I am a co-founder @CodSpeedHQ.

I took a look at your setup with CodSpeed, and saw that you are using matrices 
It means that for each run of the workflow, there will be multiple uploads of the same benchmarks, which is currently not supported by CodSpeed.

I greatly recommend that you upload only once per workflow, and if possible with the latest python version (`3.12`), so that you can benefit from [automated flamegraphs](https://docs.codspeed.io/instruments/cpu/trace-generation/).

This should do the trick:

```suggestion
    - name: Run benchmarks
      if: ${{ matrix.python == '3.12' }}
      uses: CodSpeedHQ/action@v3
      with:
        token: ${{ secrets.CODSPEED_TOKEN }}
        run: pytest tests/ --codspeed
```